### PR TITLE
Jtj 45 log entries overwiew page

### PR DESCRIPTION
**Pull request by tgi-wtag**
[JTJ-45](https://welltravel.atlassian.net/browse/JTJ-45)

This PR includes the creation of the Entries Overview page, this is where the Entries get shown and later will be selected to show in detail, edit or delete. 

https://github.com/tgi-wtag/Journey-tg/assets/141820846/ed71f651-6755-4ecf-b52d-26599fe49b97

### CODE REVIEW

File: app/models/goal.rb [hunk 1]
```diff
@@ -3,4 +3,6 @@ class Goal < ApplicationRecord
     setted: 0,
     working_on: 1,
     archived: 2
+  }
+  validates_presence_of :description
```
Comment (lra-wtag) [inline app/models/goal.rb:7]:
```suggestion
  validates :description, presence: true
```

File: app/models/journal_entry.rb [hunk 2]
```diff
@@ -1,2 +1,4 @@
 class JournalEntry < ApplicationRecord
+  validates_presence_of :title
```
Comment (lra-wtag) [inline app/models/journal_entry.rb:2]: follow the previous suggestion

File: app/models/journal_entry.rb [hunk 3]
```diff
@@ -1,2 +1,4 @@
...
+  validates_presence_of :content
```
Comment (lra-wtag) [inline app/models/journal_entry.rb:3]: follow the previous suggestion

File: app/models/task.rb [hunk 4]
```diff
@@ -3,4 +3,6 @@ class Task < ApplicationRecord
     todo: 0,
     in_progress: 1,
     finished: 2
+  }
+  validates_presence_of :title
```
Comment (lra-wtag) [inline app/models/task.rb:7]: follow the previous suggestion

**Review by lra-wtag — CHANGES_REQUESTED** [review]

File: app/models/goal.rb (hunk 1 above)
Comment (tgi-wtag) [reply app/models/goal.rb:7]: thank you for the feedback i changed it

File: app/models/user.rb [hunk 5]
```diff
@@ -1,2 +1,2 @@
 class User < ApplicationRecord
-end
+end
```
Comment (lra-wtag) [inline app/models/user.rb:2]: why the extra space after end? please remove it

File: Gemfile [hunk 6]
```diff
@@ -18,6 +18,9 @@ gem 'turbo-rails'
 
 group :development, :test do
   gem 'debug', platforms: %i[ mri mingw x64_mingw ]
+  gem 'rspec-rails', '~> 6.0'
+  gem 'factory_bot_rails', '~> 6.2'
+  gem 'rails-controller-testing', '~> 1.0'
```
Comment (chm-wtag) [inline Gemfile:23]: we should sort the gem names alphabetically

File: spec/factories/journal_entries.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :journal_entry do
+    title { 'Sample Entry1' }
+    date { Date.today }
+    content { 'This is a test entry.' }
```
Comment (chm-wtag) [inline spec/factories/journal_entries.rb:5]: we could use faker gem for dummy data

File: spec/factories/journal_entries.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
```
Comment (chm-wtag) [inline spec/factories/journal_entries.rb:1]: don't we need factories for other models?

Comment (tgi-wtag) [reply spec/factories/journal_entries.rb:1]: yes your right later i will need them but so far i haven't added any functionality for the other views so i will ad the factories when i add tests for the other controllers once their used

File: spec/factories/journal_entries.rb (hunk 7 above)
Comment (tgi-wtag) [reply spec/factories/journal_entries.rb]: thank you for the feedback i updated it

**Comment by chm-wtag** [comment]
It is a good practice to run rubocop ONLY on the code that we are currently working with, rather than the whole project/files.
Running rubocop all over the project makes the PR unnecessarily lengthy (sometimes messy as well). Could you please discard the rubocop changes from files that are not related to this PR? For example: [app/controllers/tasks_controller.rb](https://github.com/tgi-wtag/Journey-tg/pull/4/files#diff-de1aaaf7a00c963e6a5e77045d8cbdafa5ebc81e1250a7ea1862a51d824e9945) This file-change is not relevant to this PR.

Also, we don't need to put `# frozen_string_literal: true` on top of each file. We can remove this.

**Review by chm-wtag — APPROVED** [review]

**Comment by tgi-wtag** [comment]
> It is a good practice to run rubocop ONLY on the code that we are currently working with, rather than the whole project/files. Running rubocop all over the project makes the PR unnecessarily lengthy (sometimes messy as well). Could you please discard the rubocop changes from files that are not related to this PR? For example: [app/controllers/tasks_controller.rb](https://github.com/tgi-wtag/Journey-tg/pull/4/files#diff-de1aaaf7a00c963e6a5e77045d8cbdafa5ebc81e1250a7ea1862a51d824e9945) This file-change is not relevant to this PR.
> 
> Also, we don't need to put `# frozen_string_literal: true` on top of each file. We can remove this.

okey thank you for the information @chm-wtag i undid my commit and now only ran the changed files

File: .rubocop.yml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+AllCops:
+  Exclude:
+    -FrozenStringLiteralComment
```
Comment (lra-wtag) [inline .rubocop.yml:3]: add an empty line at the end of the file please

File: app/channels/application_cable/connection.rb [hunk 10]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (lra-wtag) [inline app/channels/application_cable/connection.rb:1]: please remove these comments from all files

**Review by lra-wtag — CHANGES_REQUESTED** [review]

File: app/channels/application_cable/connection.rb (hunk 10 above)
Comment (tgi-wtag) [reply app/channels/application_cable/connection.rb]: i think i got all now, thank you

**Review by lra-wtag — CHANGES_REQUESTED** [review]

File: .rubocop.yml [hunk 11]
```diff
@@ -0,0 +1,4 @@
...
+
```
[the remark this answers is not in the document]
Comment (tgi-wtag) [reply .rubocop.yml:4]: if i backspace and hit enter the cursor will be 2 tabs indented, as it is now.

**Review by lra-wtag — CHANGES_REQUESTED** [review]
@tgi-wtag can you please undo the changes from the migrations and schema. We should not alter the migrations nor the schema. Please keep it in mind

**Review by lra-wtag — APPROVED** [review]

**Pull request merged**